<launch>


    <!--General settings for the entire pipeline-->
    <!-- Frame IDs -->
    <arg name="map_frame_id"       default="vdomap"/>
    <arg name="odom_frame_id"     default="vdoodom"/>
    <arg name="base_link_frame_id" default="vdobase_link"/>

    

    <!--Settings for the visualizer. rosparam names will be set under this namespace-->

    <!-- Empty string ("") means no ground-truth available. Used for init if
        requested to do ground-truth initialization. -->
    <arg name="odometry_ground_truth_topic"       default="/tesse/odom"/>

    <!--Input gps topic name if available-->
    <arg name="gps_topic"       default=""/>
    <arg name="display_window" default="true"/>


    
    <!--Actually add the params to the paramserver under the correct namespace-->
    <group ns="vdo_pipeline">

    <group ns="visualizer">
        
        <param name="odometry_ground_truth_topic"       value="$(arg odometry_ground_truth_topic)"/>
        <param name="gps_topic" value="$(arg gps_topic)"/>
        <param name="display_window" value="$(arg display_window)"/>
                
    </group>

    <!-- Frame IDs -->
    <param name="map_frame_id"        value="$(arg map_frame_id)"/>
    <param name="odom_frame_id"      value="$(arg odom_frame_id)"/>
    <param name="base_link_frame_id"  value="$(arg base_link_frame_id)"/>


        
    </group> <!--End ns="ros_vdo_slam"-->

    <arg name="config_file" default="realtime_vdo.yaml"/>
    <rosparam command="load" file="$(find my_realtime_vdo_slam)/config/$(arg config_file)"/>

    
</launch>